addons/mail/static/tests/message/message_tests.js

```diff
@@ -42,6 +43,73 @@ QUnit.test("Start edition on click edit", async () => {
     await contains(".o-mail-Message-editable .o-mail-Composer-input", { value: "Hello world" });
 });
 
+QUnit.test("Editing message keeps the mentioned channels", async () => {
+    const pyEnv = await startServer();
+    const channelId1 = pyEnv["discuss.channel"].create({
+        name: "general",
+        channel_type: "channel",
+    });
+    const channelId2 = pyEnv["discuss.channel"].create({
+        name: "other",
+        channel_type: "channel",
+    });
+    pyEnv["mail.message"].create({
+        author_id: pyEnv.currentPartnerId,
+        body: markup(
+            `<span><a href='http://test.odoo.com/web#model='discuss.channel'&amp;id=${channelId2}' class='o_channel_redirect' data-oe-id='${channelId2}' data-oe-model='discuss.channel' target='_blank'>#other</a> hello it's me</span>`
+        ),
+        model: "discuss.channel",
+        res_id: channelId1,
+        message_type: "comment",
+    });
+    const { openDiscuss } = await start();
+    openDiscuss(channelId1);
+    await contains(".o_channel_redirect", { count: 1, text: "#other" });
+    await click(".o-mail-Message [title='Expand']");
+    await click(".o-mail-Message [title='Edit']");
+    await contains(".o-mail-Message-editable .o-mail-Composer-input", {
+        value: "#other hello it's me",
+    });
+    await insertText(".o-mail-Message .o-mail-Composer-input", "#other bye", { replace: true });
+    await click(".o-mail-Message a", { text: "save" });
+    await contains(".o-mail-Message-content", { text: "#other bye" });
+    await click(".o_channel_redirect", { text: "#other" });
+    await contains(".o-mail-Discuss-threadName", { value: "other" });
+});
+
+QUnit.test("Editing message drops the outdated channels", async () => {
+    const pyEnv = await startServer();
+    const channelId1 = pyEnv["discuss.channel"].create({
+        name: "general",
+        channel_type: "channel",
+    });
```

Comment:
Name of test is confusing. The keyword "mention" is lacking and very important to understand the purpose of test.

And also "outdated" is a poor choice of word. This is appropriate when an expiration date is obvious in the context but here this is not.

If I understand the intent of test, you want to check that clearing a channel mention doesn't mention it again? I don't really see how you could have code that's mistakenly too naïve and let this test pass... I'd drop the test if there's not a more relevant version of it.

Reply:
The point of this task is to check if entering the edit mode, when the valid mentions become non-valid, it will be cleared or not. I don't mind getting rid of this but I think it is valuable though the key function `getMentionsFromText` can be checked somewhere else.

Reply:
The current issue is that mentioned channels are lost. By "lost", the text is present but the linkify is lost. The 1st test checks that.

But if there's a channel mention, you fully edit the message so it contains no textual channel mention, and then assert there's no channel mention in it... I don't know how such a test could pass without having explicitly reached this problem.

As it stands, the test seems to exist just for the sake of existing. The scenario of the test never existed and I don't see how this could potentially happen in near future. Each test is added code maintenance, so it's not ok to add tests just for the sake of having more tests.